native_modules/NavigationApplication.java

```diff
@@ -0,0 +1,141 @@
+package com.reactnativenavigation;
+
+import android.app.Application;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityOptionsCompat;
+
+import com.facebook.react.ReactApplication;
+import com.facebook.react.ReactNativeHost;
+import com.facebook.react.ReactPackage;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.uimanager.UIManagerModule;
+import com.reactnativenavigation.bridge.EventEmitter;
+import com.reactnativenavigation.controllers.ActivityCallbacks;
+import com.reactnativenavigation.react.NavigationReactGateway;
+import com.reactnativenavigation.react.ReactGateway;
+
+import java.util.List;
+
+public abstract class NavigationApplication extends Application implements ReactApplication {
+    public final static long APP_START_TIME = System.currentTimeMillis();
```

Comment:
I think we should remove this file as it does not do anything more than adding the telemetry stuff, right?